homeassistant/components/recollect_waste/__init__.py

```diff
@@ -31,7 +31,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     async def async_get_pickup_events() -> list[PickupEvent]:
         """Get the next pickup."""
         try:
-            return await client.async_get_pickup_events()
+            # Retrieve (today-1) through to 35 days in the future, to get
+            # coverage across a full two months boundary so that no
+            # upcoming pickups are missed. The api.recollect.net base API
+            # call returns only the current month when no dates are passed.
+            # This ensures that data about when the next pickup is will be
+            # returned when the next pickup is the first day of the next month.
+            # Ex: Today is August 31st, tomorrow is a pickup on September 1st.
+            return await client.async_get_pickup_events(
+                start_date=date.today() - timedelta(days=1),
```

Comment:
Do we really need to include yesterday?

The coordinator runs once a day and sensor and calendar have the following check `if e.date >= date.today()`

Reply:
I did that to hedge on what the API would return from the upstream side. I didn't want to assume their logic would be `>=` but could be `>` in the results returned. So it could miss the "today"

Reply:
Seems to not be an issue, I can update.

```
➜  ~ curl -s https://api.recollect.net/api/places/561F80AC-DA9C-11E9-85F8-60C6F93C4E98/services/1110/events\?after\=2023-09-01\&before\=2023-10-31 |jq '.events' |grep 2023
      "connector_created": "2023-09-01"
    "day": "2023-09-01",
      "connector_created": "2023-09-08"
    "day": "2023-09-08",
      "connector_created": "2023-09-15",
    "day": "2023-09-15",
    "day": "2023-09-22",
      "connector_created": "2023-09-22",
    "day": "2023-09-29",
      "connector_created": "2023-09-29"
    "day": "2023-10-06",
      "connector_created": "2023-10-06",
      "connector_created": "2023-10-13"
    "day": "2023-10-13",
      "connector_created": "2023-10-20"
    "day": "2023-10-20",
      "connector_created": "2023-10-27"
    "day": "2023-10-27",
```